website/guides/running-terraform-in-automation.html.md

```diff
@@ -0,0 +1,279 @@
+can be used:
+
+* `terraform init -input=false`
+* `terraform apply -input=false -auto-approve=true`
+
+This variant of the `apply` command implicitly creates a new plan and then
+immediately applies it. The `-auto-approve=true` option tells Terraform not
+to require interactive approval of the plan before applying it.
+
+~> When Terraform is empowered to make destructive changes to infrastructure,
+manual review of plans is always recommended unless downtime is tolerated
+in the event of unintended changes. Use automatic apply **only** with
+non-critical infrastructure.
+
+## Testing Pull Requests with `terraform plan`
+
+`terraform plan` can be used as a way to perform certain limited verification
+of the validity of a Terraform configuration, without affecting real
+infrastructure. Although the plan step updates the state to match real
+resources, thus ensuring an accurate plan, the updated state is _not_
+persisted, and so this command can safely be used to produce "throwaway" plans
+that are created only to aid in code review.
+
+When implementing such a workflow, hooks can be used within the code review
+tool in question (for example, Github Pull Requests) to trigger an orchestration
+tool for each new commit under review. Terraform can be run in this case
+as follows:
+
+* `terraform plan -input=false`
+
+As in the "main" workflow, it may be necessary to provide `-var` or `-var-file`
+as appropriate. The `-out` option is not used in this scenario because a
+plan produced for code review purposes will never be applied. Instead, a
+new plan can be created and applied from the primary version control branch
+once the change is merged.
+
+~> Beware that passing sensitive/secret data to Terraform via
+variables or via environment variables will make it possible for anyone who
+can submit a PR to discover those values, so this flow must be
+used with care on an open source project, or on any private project where
+some or all contributors should not have direct access to credentials, etc.
+
+## Multi-environment Deployment
+
+Automation of Terraform often goes hand-in-hand with creating the same
+configuration multiple times to produce parallel environments for use-cases
+such as pre-release testing or multi-tenant infrastructure. Automation
+in such a situation can help ensure that the correct settings are used for
+each environment, and that the working directory is properly configured
+before each operation.
+
+The two most interesting commands for multi-environment orchestration are
+`terraform init` and `terraform workspace`. The former can be used with
+additional options to tailor the backend configuration for any differences
+between environments, while the latter can be used to safely switch between
+multiple states for the same config stored in a single backend.
+
+Where possible, it's recommended to use a single backend configuration for
+all environments and use the `terraform workspace` command to switch
+between workspaces:
```

Comment:
If I understand correctly we're suggesting the exact opposite of https://www.terraform.io/docs/state/environments.html#best-practices here?